docs/source/sparse.rst

```diff
@@ -167,6 +171,160 @@ receiving a particular layout. We are working on an API to control the result la
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.],
+	    [0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1., 0., 0., 1., 1.]],
+	   device='cuda:0', dtype=torch.float16)
+    >>> a = to_sparse_semi_structured(a)
+    SparseSemiStructuredTensor(shape=torch.Size([16, 16])
+     metadata=tensor([[-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370],
+	    [-4370]], device='cuda:0', dtype=torch.int16)
+     values=tensor([[1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.],
+	    [1., 1., 1., 1., 1., 1., 1., 1.]], device='cuda:0',
+	   dtype=torch.float16))
+
+Semi-Structured Sparse Operations
+-------------------------------------------
+
+We currently support the following ops:
+
+- torch.addmm(bias, dense, sparse)
+- torch.addmm(bias, sparse, dense)
+- torch.mm(dense, sparse)
+- torch.mm(sparse, dense)
+- aten.linear.default(dense, sparse, bias)
+- aten.t.default(sparse)
+- aten.t.detach(sparse)
+
+To use these ops, simply pass the output of ``to_sparse_semi_structured(tensor)``  instead of using ``tensor`` once your tensor has 0s in a semi-structured sparse format, like this:
+
+    >>> a = torch.Tensor([0, 0, 1, 1]).tile((64, 16)).half().cuda()
+    >>> b = torch.rand(64, 64).half().cuda()
+    >>> dense = torch.mm(a, b)
+    >>> a = to_sparse_semi_structured(a)
+    >>> sparse = torch.mm(a, b)
+    >>> torch.allclose(dense, sparse))
+    True
+
+
+Accelerating nn.Linear with semi-structured sparsity
+----------------------------------------------------
+You can accelerate the linear layers in your model if the weights are already semi-structured sparse with just a few lines of code:
+
+    >>> input = torch.rand(64, 64).half().cuda()
+    >>> mask = torch.Tensor([0, 0, 1, 1]).tile((64, 16)).cuda()
+    >>> linear = nn.Linear(64, 64).half().cuda()
+    >>> linear.weight.data*=mask
```

Comment:
Super nit, but I think maybe masked_fill_(mask, 0) would be more efficient.

I think you could even [use broadcasting](https://pytorch.org/docs/master/generated/torch.Tensor.masked_fill_.html?highlight=masked_fill_#torch.Tensor.masked_fill_) and use view like this: ```linear.weight.data.view(-1, 4).masked_fill_(torch.tensor([1, 1, 0, 0], device=torch.device('cuda')), 0)```